features/frameworks/mbed-coap/mbed-coap/sn_config.h

```diff
@@ -68,6 +68,23 @@
 #define SN_COAP_MAX_BLOCKWISE_PAYLOAD_SIZE              0  /**< Must be 2^x and x is at least 4. Suitable values: 0, 16, 32, 64, 128, 256, 512 and 1024 */
 #endif
 
+/**
+ * \def SN_COAP_CONSTANT_NEEDED_SIZE
+ * \brief Avoid needed size calculations
+ * If this is defined, sn_coap_builder_calc_needed_packet_data_size always returns that value,
+ * saving a lot of calculation code, at the cost of outgoing TX buffers being oversized, and
+ * with danger of them being undersized.
+ *
+ * sn_coap_builder_payload_build does not have any size input to limit its output, so it is
+ * always wise for users to assert that it has not output more than the size returned by
+ * sn_coap_builder_calc_needed_packet_size, whether this option is defined or not.
+ */
+#ifdef MBED_CONF_MBED_CLIENT_SN_COAP_CONSTANT_NEEDED_SIZE
```

Comment:
I think this should have been defined in mbed-coap's mbed_lib.json to have a default value (null). Now this is a client related define.

Reply:
But it can be changed as a separate PR.